lib/cellect/server/workflow.rb

```diff
@@ -41,23 +46,39 @@ def initialize(name, pairwise: false, prioritized: false)
         self.users = { }
         self.pairwise = !!pairwise
         self.prioritized = !!prioritized
-        self.subjects = set_klass.new
+        self.subjects ||= set_klass.new
+        self.state = :initializing
+        self.can_reload = false
       end
 
       # Loads subjects from the adapter
       def load_data
-        self.state = :initializing
-        self.subjects = set_klass.new
-        Cellect::Server.adapter.load_data_for(name).each do |hash|
-          subjects.add hash['id'], hash['priority']
-        end
+        return if self.state == :ready
+        self.state = :loading
+        load_adapter_data(self.subjects)
+        reset_can_reload_timer
         self.state = :ready
       end
 
+      # Reloads subjects from the adapter
+      def reload_data
+        if can_reload_data?
+          self.can_reload = false
+          self.state = :reloading
+          new_data = self.subjects.class.new
```

Comment:
Good question, reload_data is async via the api call [here](https://github.com/zooniverse/Cellect/blob/9db682078b1573f4142cfa2f8d27609e784f9c7c/lib/cellect/server/api.rb#L59) whereas `load_data` is not async (currently). It'll make a new set of subjecs and switch them in after successful reloading is complete.

Rather it creates a supervised set of worfklows via `self.[]=` and the `Cellect::Server::Adapters:Default` adapter `worfklow_for` method, all called via the api if not already setup [here](https://github.com/zooniverse/Cellect/blob/9db682078b1573f4142cfa2f8d27609e784f9c7c/lib/cellect/server/api/helpers.rb#L6). It will load the data in band after initialising and registering the workflow for selection endpoints. Data will trickle into the set gradually as each page populates instead of a bulk switch like `reload_workflow`. 

The setup / init of workflow & user data is a bit convoluted but I'd prefer to work on that in another PR.